plugins/modules/identity/keycloak/keycloak_user.py

```diff
@@ -0,0 +1,530 @@
+    client_consents:
+        description:
+            - The client consents.
+        aliases:
+            - clientConsents
+        type: list
+        elements: dict
+        suboptions:
+            client_id:
+                description:
+                    - The client ID.
+                aliases:
+                    - clientId
+                type: str
+            granted_client_scopes:
+                description:
+                    - The granted client scopes.
+                aliases:
+                    - grantedClientScopes
+                type: list
+                elements: str
+    credentials:
+        description:
+            - The credentials.
+        type: list
+        elements: dict
+        suboptions:
+            priority:
+                description:
+                    - The priority.
+                type: int
+            type:
+                description:
+                    - The type.
+                type: str
+            temporary:
+                description:
+                    - The temporary.
+                type: bool
+            value:
+                description:
+                    - The value.
+                type: str
+    disableable_credential_types:
+        description:
+            - The disableable credential types.
+        aliases:
+            - disableableCredentialTypes
+        type: list
+        elements: str
+    email:
+        description:
+            - The email.
+        type: str
+    email_verified:
+        description:
+            - The email verified.
+        aliases:
+            - emailVerified
+        type: bool
+    enabled:
+        description:
+            - The enabled.
+        type: bool
+    first_name:
+        description:
+            - The first name.
+        aliases:
+            - firstName
+        type: str
+    groups:
+        description:
+            - The group pathes.
+        type: list
+        elements: str
+    id:
+        description:
+            - The unique identifier for this user.
+            - This parameter is not required for updating or deleting a user but
+              providing it will reduce the number of API calls required.
+        type: str
+    last_name:
+        description:
+            - The last name.
+        aliases:
+            - lastName
+        type: str
+    not_before:
+        description:
+            - The not before.
+        aliases:
+            - notBefore
+        type: int
+    origin:
+        description:
+            - The origin.
+        type: str
+    realm_roles:
+        description:
+            - The realm roles.
+        aliases:
+            - realmRoles
+        type: list
+        elements: str
+    required_actions:
+        description:
+            - The required actions.
+        aliases:
+            - requiredActions
+        type: list
+        elements: str
+    self:
+        description:
+            - The self.
+        type: str
```

Comment:
Isn't `self` a reserved word in python? I'd suggest we should be using a different variable name within the code. (e.g. _self`.   - However, I'm not actually familiar with what the property, `self` is for a user.

Reply:
In Python code `self` shouldn't be used, but as an Ansible module option it is valid.

Reply:
I've searched for where `self` has been used and I can't find it (apart from being declared as a parameter)

Reply:
It's used here (indirectly): https://github.com/ansible-collections/community.general/pull/3996/files#diff-27484ef56e5723b6d2afcf5d3090be54364c2bbbc6102b7bd57c037d0a0f6f87R430

Reply:
Thanks, I can't seem to see what it is used for from the Keycloak API documentation.

Reply:
I use `self` is just because of this document: https://www.keycloak.org/docs-api/15.0/rest-api/index.html#_userrepresentation. I'm not sure what it is and its usage but if the official document has it, I think I should also have it.

Reply:
I wish the API documentation would also have some basic description, and not just type + whether it's optional... I also tried grepping the keycloak codebase for it, and only found that it exists, but no actual use of it :shrug:

Reply:
@fynncfchen I found the same link you sent and was disappointed that the documentation was incomplete

Regarding this conclusion:
> I'm not sure what it is and its usage but if the official document has it, I think I should also have it.

I actually disagree with this, (though not strongly).  If we don't know what something is, and we don't need to send it or modify it, and we don't know what it is, I'm not sure why we should have it - (how can we know that our implementation actually works).  Is there's a chance that it's a keycloak internal property that accidentally got exposed on the API documentation page?

Reply:
After grepping around some more in the Git history, it looks like this field (at least in some other contexts) was used to store the URI of the object it is part of (https://github.com/keycloak/keycloak/commit/7a1c825dcbeaa939a744f8b77facef25782501da#diff-87fb13ab10accda0d77c1c824d82122c69abc470b749fdfa621c32d2731df71dR140; in that context, it was eventually removed in https://github.com/keycloak/keycloak/commit/11559cba50c594532aa6aaac6aa3bf647319eacd#diff-d720c1c4687daf1792fb3e053e1762149ed990540c5fe0cd86396996e8c74aebL71). Maybe the remaining `self` fields are just an oversight?

In any case, not exposing this is probably the best solution. If anyone figures out what it is useful for, it's very easy to add it later.

Reply:
Sure, I agree with both of you, I'll remove it, thanks!